bson/setter_getter.go

```diff
@@ -32,14 +28,14 @@ import (
 //
 //	type MyString string
 //
-//	func (s *MyString) SetBSON(raw bson.RawValue) error {
+//	func (s *MyString) SetBSON(raw RawValue) error {
 //	    return raw.Unmarshal(s)
 //	}
 type Setter interface {
-	SetBSON(raw bson.RawValue) error
+	SetBSON(raw RawValue) error
 }
```

Comment:
I'm concerned merging the `mgocompat` package into the `bson` package may create some namespacing problems. I know that was part of the original scope, but looking at the results, it's not clear that it makes sense. Since `mgocompat` only configures alternative encode/decode behaviors via a `Registry` and is not used by any of the core BSON logic, there's little risk of import cycles. Should we keep `mgocompat` a separate package?

Reply:
I wonder how long we will support the compatibility with mgo?

Reply:
As far as mgo BSON marshal/unmarshal compatibility, I think we need to support it indefinitely. Customers with data in a database that was marshaled with mgo may continue to need unmarshaling compatibility unless they rewrite all incompatible documents with the Go Driver.